From 3f99de7350d5a181aba9e06d9b711bc967385001 Mon Sep 17 00:00:00 2001
From: Monika Korada <quic_koramoni@quicinc.com>
Date: Mon, 10 Jun 2024 12:17:05 +0530
Subject: [PATCH] KW: wifi: ath12k: Fix NULL pointer access in ieee80211_do_stop function

Currently, we are dereferencing sdata->dev without a NULL check in
ieee80211_do_stop function. Fix this issue by adding a NULL check before
accessing the sdata->dev.

Patch-dependency: none
Patch-work: none
Fixes: d2859df5e7f0 (cfg80211/mac80211: DFS setup chandef for cac event)
Note: ath.git commit id is mentioned in the Fixes tag

Signed-off-by: Monika Korada <quic_koramoni@quicinc.com>
Signed-off-by: Naveen S <quic_naves@quicinc.com>
---
 net/mac80211/iface.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -579,9 +579,10 @@ static void ieee80211_do_stop(struct iee
 		mutex_lock(&local->mtx);
 		ieee80211_link_release_channel(&sdata->deflink);
 		mutex_unlock(&local->mtx);
-		cfg80211_cac_event(sdata->dev, &chandef,
-				   NL80211_RADAR_CAC_ABORTED,
-				   GFP_KERNEL, 0);
+		if (sdata->dev)
+			cfg80211_cac_event(sdata->dev, &chandef,
+					   NL80211_RADAR_CAC_ABORTED,
+					   GFP_KERNEL, 0);
 	}
 
 	if (sdata->vif.type == NL80211_IFTYPE_AP) {
